torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -96,12 +96,19 @@ TensorView::TensorView(
         // dim;
         contig_info[index] = (index == tensor_type->dim().value() - 1);
       } else {
-        // check the neighboring faster dimension;
-        if (auto left_index_opt =
-                tensor_type->stride_properties()[static_cast<int>(i) - 1]
-                    ->stride_index_) {
-          // collapse if two axes are neighboring in both sizes & stride_index;
-          contig_info[index] = (left_index_opt.value() == (index + 1));
+        // check the neighboring faster dimension, collapse if two axes are
+        // neighboring in stride_index;
+        auto left_index_opt =
+            tensor_type->stride_properties()[static_cast<int>(i) - 1]
+                ->stride_index_;
+        if (left_index_opt.has_value() &&
+            left_index_opt.value() == (index + 1)) {
```

Comment:
how come you are collapsing based only on order of the strides, and not strides and sizes? What if tensor is not memory dense?

Reply:
Totally not expecting to run into you here :)

But we did peek contiguous flag, which is supposed to check stride/size to ensure neighboring dimensions are memory dense: https://github.com/csarofeen/pytorch/pull/1519/files/0372548daf10ace4da6dddf1c1fc5c6ca8ce4b5c#diff-28d672948f070384f5d8c15ec70180f5ac1b7632d713af96dec1634fbdbb89ceR91

Reply:
Ok, then my question is should `contiguous_` value be fixed in `computeStrideProps`? There's
```
        contiguous_ = strides[stride_indices[i]] == 1 ||
            (strides[stride_indices[i]] != 0 &&
             strides[stride_indices[i]] ==
                 strides[stride_indices[i - 1]] * sizes[stride_indices[i - 1]]);
 ```
 why is stride of 1 enough to mark something contiguous? I'd understand if size of 1 was.

Reply:
> why is stride of 1 enough to mark something contiguous? I'd understand if size of 1 was.

This is a tricky topic. I think the assumption is that we won't have overlapped memory, so `stride == 1` means that current dimension would be the fastest dimension with non-trivial length, while implicitly broadcasted dimensions are technically faster but not collapsible.

In reality I don't think we are efficiently leveraging this yet....

Reply:
Why do you have an assumption that you won't have overlapped memory?

Reply:
That's a fair ask. we (nvfuser) don't have determined outputs for overlapped tensor, because of the racing conditions on write. So we don't care much about it. :stuck_out_tongue: 

In case of contiguity map for overlapped memory, it's also hard/tricky to figure out which dimensions are overlapping in the general case. reference:  https://github.com/pytorch/pytorch/blob/0be36d798ba959bfda6c448fc4832b5691df6e61/aten/src/ATen/MemoryOverlap.h#L12-L19
I do agree that we can be a little bit more cautious/conservative on this and mark all contiguity as false for YES/TOO_HARD cases.

Reply:
I understand, but in this case contiguity map was wrong for an input tensor, not output tensor, and input tensors are allowed to be non-overlapping. General overlapping is hard, but contiguity isn't.